src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressByMessageFilterTest.java

```diff
@@ -0,0 +1,164 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.filters;
+
+import static com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck.MSG_METHOD;
+import static com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck.MSG_VARIABLE;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.TreeWalker;
+import com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+
+public class SuppressByMessageFilterTest extends AbstractModuleTestSupport {
+
+    @Override
+    protected String getPackageLocation() {
+        return "com/puppycrawl/tools/checkstyle/filters/suppressbymessagefilter";
+    }
+
+    @Test
+    public void testDefault() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(SuppressByMessageFilter.class);
+
+        final String[] expectedViolations = {
+            "14:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "a", ""),
+            "15:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "b", ""),
+            "16:9: " + getCheckMessage(RequireThisCheck.class, MSG_METHOD, "foo", ""),
+            "20:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "c", ""),
+        };
+
+        verifySuppressed(config, getPath("InputSuppressByMessageFilter.java"),
```

Comment:
You'll probably be requested to convert this to our BDD testing. I would look over an existing filter on what it will ask for.